src/Core/tests/DeviceTests/Handlers/Window/WindowHandlerTests.cs

```diff
@@ -54,18 +49,96 @@ public async Task InitialPositionsAreTakenIntoAccount()
 			{
 				// Just let things settle for good measure
 				await Task.Delay(100);
+
+#if MACCATALYST
+				// these are updated by the OS
+				Assert.True(window.Width > 0, $"Expected Width to be >= 0, but was {window.Width}");
+				Assert.True(window.Height > 0, $"Expected Height to be >= 0, but was {window.Height}");
+				// these are not available from the OS...
+				Assert.True(window.X == 0, $"Expected X to be == 0, but was {window.X}");
+				Assert.True(window.Y == 0, $"Expected Y to be == 0, but was {window.Y}");
+#elif WINDOWS
+				Assert.Equal(200, window.Width);
+				Assert.Equal(500, window.Height);
+				Assert.Equal(0, window.X);
+				Assert.Equal(500, window.Y);
+#endif
+			});
+		}
+
+		[Fact]
+		public async Task UpdatedPositionsAreTakenIntoAccount()
+		{
+			var window = new Window(new NavigationPage(new ContentPage()));
+
+			await RunWindowTest(window, async handler =>
+			{
+				var currentFrame = new Rect(window.X, window.Y, window.Width, window.Height);
+
+				// Just let things settle for good measure
+				await Task.Delay(100);
+
+				window.Width = 200;
+				window.Height = 500;
+				window.X = 0;
+
+				// Just let things settle for good measure
+				await Task.Delay(100);
+
+#if MACCATALYST
+				// Mac Catalyst does not support this operation, so it should never change
+				Assert.Equal(currentFrame, new Rect(window.X, window.Y, window.Width, window.Height));
+#elif WINDOWS
 				Assert.Equal(200, window.Width);
 				Assert.Equal(500, window.Height);
 				Assert.Equal(0, window.X);
 				Assert.Equal(500, window.Y);
+#endif
 			});
 		}
 
-		[Fact(
+		[Fact(Skip = "This is the OG issue")]
+		public async Task ChangingSizeWhileChangingSize()
```

Comment:
This test is the one that a user may write to repro the issue. However, it is still looping.